RBAC API for Metrics Access (#1)
stolostron/rbac-api-utils: README.md

# rbac-api-utils

> A library for fine-grained multi cluster rbac api

Reviewer: Are there any usage scenarios that could be provided beyond just letting developers know they need to run `make test`.  So something like:

This is an example spec of a `Role` and `RoleBinding` object that can be used to provide fine grained access to user `fred` for managed cluster `dev-cluster` and namespace `app-for-fred`.

```
role and role binding here
```

Author: @gparvin  I added doc for the function inline with details of the input and out params.
I also added an example scenario with role, role bindings and results for sample invocations of the API
Could you pls take a look again

Reviewer: Thanks!

# RBAC API Utils
[![License](https://img.shields.io/:license-apache-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0.html)

A library for multi-cluster RBAC API. Access control in OCM is based on RBAC contructs  provided by Kubernetes,

> this library provides utiilies and  wrappers around the kubernetes RBAC API for convenience and 

Reviewer 2: ```suggestion
this library provides utiilies and  wrappers around the Kubernetes RBAC API for convenience and 
```

to achieve finer grained access control.

> Go to the [Contributing guide](CONTRIBUTING.md) to learn how to get involved.

Reviewer 2: Did you commit `CONTRIBUTING.md`?

> - The `rbac-api-utils` repository is part of the `open-cluster-management`

Reviewer 2: I'm not sure that this is true since it's in Stolostron.

Author: I wondered about this but non OCM component Repos like Cert, Iam or Search etc  all seemed to have the same text.
I will just remove the entire statement

  community. For more information, visit:
  [open-cluster-management.io](https://open-cluster-management.io).

## Usage

To get started, create an instance of the AccessReviewer as below:

```go

import (
  "github.com/stolostron/rbac-api-utils/rbac"
)

// Create an instance of AccessReviewer with the KubeConfig for the target cluster
accessReviewer := rbac.NewAccessReviewer(myTargetKubeConfig, nil)

```

> See [here](./pkg/rbac/rbac.go/#L49) for more information on the parameters for creation of a AccessReviewer

Reviewer 2: ```suggestion
See [here](./pkg/rbac/rbac.go/#L49) for more information on the parameters for creation of an `AccessReviewer`.
```

### Supported API

> **GetMetricsAccess** returns the  managedclusters and namespaces on the managedclusters for which the user has access to view observability metrics. See [here](./pkg/rbac/rbac.go/#L121) for details on the input parameters and results.

Reviewer 2: I think we would want to use the English term of `managed clusters` through this section since you aren't talking about the Kubernetes API objects.

OCM Observability gathers  metrics from the managed clusters and stores them for viewing on the Hub. Users can be given access to view metrics for specific namespaces on specific managedclusters.

> Consider the scenario where an Application "Blue" is deloyed to namespaces blue1 and blue2 on ManagedClusters  devcluster1 and devcluster2. Inorder to give the Blue admins  access to view Blue metrics, the following clusterroles are set

Reviewer 2: ```suggestion
Consider the scenario where an application "Blue" is deployed to namespaces blue1 and blue2 on managed clusters  devcluster1 and devcluster2. In order to give the Blue admins  access to view Blue metrics, the following cluster roles are set
```

> ```

Reviewer 2: ```suggestion
```yaml
```
